File: AssetCoreSol/ClientApp/src/app/services/assetModel.ts
import { Category } from "../models/category";
import { Department } from "../models/department";
import { Status } from "../models/status";
import { Asset } from "./asset";
import { Employee } from "../models/employee";
import { AssetAuditLog } from "../models/assetAuditLog";

 export class AssetModel {

  asset: Asset = new Asset; //need to instantiate here
  assetList: Asset[];
  categoryList: Category[];
  statusList: Status[];
  departmentList: Department[];
  employeeList: Employee[];
  assetAuditLogList: AssetAuditLog[];
}

File: DAL/Models/AssetModel.cs
using DAL.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace DAL.Models
{
    public class AssetModel
    {
        public Asset Asset { get; set; }
        public IEnumerable<AssetCategory> CategoryList { get; set; }
        public IEnumerable<Asset> AssetList { get; set; }
        public IEnumerable<Status> StatusList { get; set; }
        public IEnumerable<Department> DepartmentList { get; set; }
        public IEnumerable<Employee> EmployeeList { get; set; }
    }
}
